src/App.tsx

```diff
@@ -1,15 +1,66 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const isFinded = (whereSearch: string[], whatSearch: string[]) => {
+    for (let t = 0; t <= whereSearch.length; t += 1) {
+      if (whereSearch[t] === whatSearch[0]) {
+        for (let i = 0; i <= whatSearch.length; i += 1) {
+          if (i === whatSearch.length) {
+            return true;
+          }
+
+          if (whereSearch[t + i] !== whatSearch[i]) {
+            break;
+          }
+        }
+      } else if (t === whereSearch.length) {
+        return false;
+      }
+    }
+
+    return false;
+  };
+
+  useEffect(() => {
+    const filterInputAsk = query.trim().toLowerCase().split('');
+    const isClear = query === '';
+
+    if (isClear) {
+      setVisibleMovies(moviesFromServer);
+    } else {
+      const visibleSingleMovies = moviesFromServer.filter((oneOfFilms) => {
+        const titleFormatText = oneOfFilms.title.toLowerCase().split('');
+        const descrFormatText = oneOfFilms.description
+          .toLowerCase()
+          .split('');
+
+        return (
+          isFinded(titleFormatText, filterInputAsk)
```

Comment:
Why don't you use `includes` method but `for` cycle?
```
const preparedTitle = title.toLowerCase();
const preparedDescription = description.toLowerCase();
const preparedQuery = query.trim().toLowerCase();

return preparedTitle.includes(preparedQuery) || preparedDescription.includes(preparedQuery);
```

Reply:
I dont shure about this variant, but my cycle can find overlap. For example, if I write "each", it view form with "each, reach,  eached" etc. But many form find only full word.

Reply:
Thanks

Reply:
Your version simply :)